### PR TITLE
Fix return header for MD

### PR DESCRIPTION
**Pull request by tahsinaislam**
## Link to pivotal/JIRA issue
https://codeforamerica.atlassian.net/browse/FYST-1303
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

## What was done?
- Looked at xsd validations for return headers for all states
- Adjusted formatting for md specific rules for street addresses and length of city names

## How to test?
- Added relevant unit tests and fixed ones with the wrong rules

### CODE REVIEW

File: app/models/state_file_base_intake.rb [hunk 1]
```diff
@@ -246,6 +246,10 @@ def extract_apartment_from_mailing_street?
     false
   end
 
+  def city_name_length_20?
+    true
```
Comment (arinchoi03) [inline app/models/state_file_base_intake.rb:250]: should this be false?

**Review by arinchoi03 — COMMENTED** [review]
real quick q

File: spec/lib/submission_builder/ty2022/states/return_header_spec.rb [hunk 2]
```diff
@@ -222,6 +222,36 @@
...
+    end
+
+    StateFile::StateInformationService.active_state_codes.without("md").each do |state_code|
+      context "if state is not MD" do
+        let(:intake) { create "state_file_#{state_code}_intake".to_sym }
+        let(:submission) { create(:efile_submission, data_source: intake) }
+        let(:doc) { SubmissionBuilder::ReturnHeader.new(submission).document }
+
+        it "truncates city name to 22 characters" do
+          expect(doc.at("USAddress CityNm").text.length).to be <= 22
```
Comment (arinchoi03) [inline spec/lib/submission_builder/ty2022/states/return_header_spec.rb:239]: length should actually be 22

File: spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb [hunk 3]
```diff
@@ -114,8 +114,8 @@
         end
 
         it 'truncates mailing address under 35 characters' do
-          expect(xml.at("MarylandAddress AddressLine1Txt").text.length).to be <= 35
-          expect(xml.at("MarylandAddress AddressLine1Txt").text).to eq('211212 SUBDIVISION DRIVE POST OFFIC')
+          expect(xml.at("MarylandAddress AddressLine1Txt").text.length).to be <= 30
```
Comment (arinchoi03) [inline spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb:117]: should be 30

File: spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb [hunk 4]
```diff
@@ -100,7 +100,7 @@
         end
 
         it 'puts apartment on second address line' do
-          expect(xml.at("MarylandAddress AddressLine1Txt").text.length).to be <= 35
+          expect(xml.at("MarylandAddress AddressLine1Txt").text.length).to be <= 30
```
Comment (arinchoi03) [inline spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb:103]: should be 30

File: spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb [hunk 5]
```diff
@@ -114,8 +114,8 @@
         end
 
         it 'truncates mailing address under 35 characters' do
```
Comment (arinchoi03) [inline spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb:116]: update this description to 30 characters?

File: spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb (hunk 3 above)
Comment (tahsinaislam) [reply spec/lib/submission_builder/ty2024/states/md/documents/md502_spec.rb:117]: it is 30?

**Review by arinchoi03 — APPROVED** [review]
yay ✨

**Pull request merged**